proxy/src/main/java/net/md_5/bungee/netty/HandlerBoss.java

```diff
@@ -128,6 +128,14 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 {
                     handler, limiter.getCounter(), limiter.getDataCounter()
                 } );
+
+                // somehow sometimes if you spam big packets fast enough the channel will not be closed immediately
+                // it seems that this only happens on linux, the PacketLimiter will trigger a kick multiple times
+                // but the connection will still be active
+                if ( channel.getHandle().isActive() )
+                {
+                    ctx.close();
```

Comment:
if there is a way we can clear the tcp read buffer with netty this could be removed.

setAutoRead(false) generally fixed the issue but there will still be up to 8 packets read after the setAutoRead(false) and close call. the channelInactive will be called after all packets are handled.

The isActive() checks will not return true in the packet handlers